File: MicroservicesArchitecture/FoodDeliveryApp.Frontend/src/app/services/payment-service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http'
import { Payment } from '../models/payment.type';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {
  private http = inject(HttpClient);
  private apiUrl = 'https://localhost:7290/api/Payments';

  getPaymentById = (id: string) => {
    const url = `${this.apiUrl}/${id}`;
    return this.http.get<Payment>(url);
  }

  getPaymentsByUserId = (userId: string) => {
    const url = `${this.apiUrl}/user/${userId}`;
    return this.http.get<Array<Payment>>(url);
  }

  createPayment = (payment: Payment) => {
    return this.http.post(this.apiUrl, payment);
  }

  updatePayment = (payment: Payment) => {
    const url = `${this.apiUrl}/${payment.id}`;
    return this.http.put(url, payment);
  }

  deletePayment = (payment: Payment) => {
    const url = `${this.apiUrl}/${payment.id}`;
    return this.http.delete(url);
  }
}

File: MicroservicesArchitecture/FoodDeliveryApp.Backend/PaymentService.API/Services/PaymentService.cs
using PaymentService.API.Data;
using PaymentService.API.Models;

namespace PaymentService.API.Services
{
    public class PaymentService: IPaymentService
    {
        private readonly IPaymentRepository _paymentRepository;

        public PaymentService(IPaymentRepository paymentRepository)
        {
            _paymentRepository = paymentRepository;
        }
        public Payment getPaymentById(int id)
        {
            var payment = _paymentRepository.getPaymentById(id);
            if(payment == null)
            {
                throw new ArgumentException("Payment not found!");
            }
            return payment;
        }

        public IEnumerable<Payment> getPaymentsByUserId(int userId)
        {
            return _paymentRepository.getPaymentsByUserId(userId);
        }

        public void createPayment(Payment payment)
        {
            _paymentRepository.createPayment(payment);
        }

        public void updatePayment(Payment payment)
        {
            var existingPayment = _paymentRepository.getPaymentById(payment.Id);
            if (existingPayment == null)
            {
                throw new ArgumentException("Payment not found!");
            }
            else
            {
                existingPayment.OrderId = payment.OrderId;
                existingPayment.UserId = payment.UserId;
                existingPayment.Amount = payment.Amount;
                existingPayment.Method = payment.Method;
                _paymentRepository.updatePayment(existingPayment);
            }
        }

        public void deletePayment(int id)
        {
            var payment = _paymentRepository.getPaymentById(id);
            if(payment == null)
            {
                throw new ArgumentException("Payment not found!");
            }
            _paymentRepository.deletePayment(id);
        }
    }
}
